src/librustc_resolve/lib.rs

```diff
@@ -484,8 +484,8 @@ type BindingMap = FxHashMap<Ident, BindingInfo>;
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
 enum PatternSource {
     Match,
-    IfLet,
-    WhileLet,
+    // FIXME(54883): Consider fusing with `Let` below once let-statements support or-patterns.
```

Comment:
Looks like they can already be merged, the only difference between them is unreachable `span_bug!`.

Reply:
Fixed (locally).